SQLite/Unit 1, Lesson 2.5 Airbnb.sql

```diff
@@ -0,0 +1,61 @@
+-- This file contains a set of queries and explanations for exploring the Airbnb database of listings and reviews for Vancouver, B.C. It addresses a few questions: 
+
+-- 1. What's the most expensive listing? What else can you tell me about the listing?
+-- 2. What neighborhoods seem to be the most popular?
+-- 3. What time of year is the cheapest time to go to your city? What about the busiest?
+
+-- The first query returns all columns for the row containing the maximum value on price.
+
+SELECT
+	MAX(price),
+	*
+FROM
+	listings
+
+-- This returns a listing whose price is $9,999 per night for a 2BR. Such a high price for an unremarkable two bedroom listing is suspect, and the 9999 may in fact represent a missing field. The query was then revised to elicit the maximum price less than 9999, which returns a more plausible entry.
+
+SELECT
+	MAX(price),
+	*
+FROM
+	listings
+WHERE 
+	price <9999
+  
+-- The revised query returns a listing for a "Luxurious 95' yacht w 22'beam" priced at $5,999 per night. The yacht is located in the Fairview neighborhood and available 364 days per year. The property has never been reviewed, so we cannot get more information about its popularity by joining the reviews table. The host, Francesco, has 2 listings, which suggests that the yacht is not his primary residence. 
+
+-- The second query looks into neighborhood popularity. There are a few ways to do this, but the simplest strategy given the information we have is probably to count the number of bookings per neighborhood. Following Inside Airbnb's San Francisco model, let's assume a review rate of 50%, so the number of reviews can serve as a proxy for the number of bookings. We can then count the number of reviews by neighborhood, and see which neighborhoods have the highest counts. 
+
+SELECT
+	neighbourhood,
+	SUM(number_of_reviews) total_reviews
+FROM
+	listings
+GROUP BY neighbourhood
+ORDER BY total_reviews DESC
+
+-- Not surprisingly, the affluent and touristy Downtown, West End, and Kitsilano neighborhoods are the most popular, in terms of number of reviews. Tourists visiting the city tend to want to stay in a central location near major attractions. Given their affluence, these areas are also quite densely populated, so we would expect to find many hosts renting properties that are not their primary residences, which is common in Airbnb listings. 
+
+-- To find out the cheapest time of year, we might look at the average listing price by month. First, the type of the price column must be cast as a double, and the dollar sign removed. The data type problem persists, even when the '$' is removed and the price column is explicitly cast as a double. The AVG function continues to return 0s. 
+
+SELECT
+	TRIM(price, '$'),
+	AVG(cast(price as double)) as avg_price,
```

Comment:
the problem is that SQL is treating `TRIM(price, '$')`and `AVG(cast(price as double))` as independent commands. so you end up with a column without $, and in the second column cast $123 as a double.